.devcontainer/.gdbinit

```diff
@@ -3,3 +3,23 @@
 # actually also works when debugging with vscode. Providing nice tools
 # to understand the internal datastructures we are working with.
 source /root/gdbpg.py
+
+# when debugging postgres it is convenient to _always_ have a breakpoint
+# trigger when an error is logged. Because .gdbinit is sourced before gdb
+# is fully attached and has the sources loaded. To make sure the breakpoint
+# is added when the library is loaded we temporary set the breakpoint pending
+# to on. After we have added out breakpoint we revert back to the default
+# configuration for breakpoint pending.
+# The breakpoint is hard to read, but at entry of the function we don't have
+# the level loaded in elevel. Instead we hardcode the location where the
+# level of the current error is stored. Also gdb doesn't understand the
+# ERROR symbol so we hardcode this to the value of ERROR. It is very unlikely
+# this value will ever change in postgres, but if it does we might need to
+# find a way to conditionally load the correct breakpoint.
+# Lastly the breakpoint doesn't show up in vscode :(, to remove the breakpoint
+# you can use the command `-exec delete <id>` in the vscode debug console. The
+# id is the number in the first column of the breakpoint list which you can
+# show with the command `-exec info break`.
```

Comment:
It might be good to print (some of) this info, so that people can remove it even if they don't read .gdbinit.

Reply:
I like the idea, do you have any idea on how to do so?

I expect you want to print this in the gdb console. The console is pretty verbose when attaching to a running citus backend due to many .so files being loaded.

We van have a look together later today to understand the value of it.

Reply:
<img width="1351" alt="image" src="https://github.com/citusdata/citus/assets/273145/4dd0b267-1f44-4d14-91f5-739780aa0267">

it looks like this now (when scrolling up)